### PR TITLE
Clean up repetitive code in CraftTreePatcher.cs

### PR DESCRIPTION
**Pull request by RamuneNeptune**
### Changes made in this pull request
  - Cut down on repetitive code
 
### Breaking changes
  - Nada. Nil. Zilch. None.

### CODE REVIEW

**Comment by RamuneNeptune** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by RamuneNeptune** [comment]
Don't do anything with this yet, gotta test it

File: Nautilus/Patchers/CraftTreePatcher.cs [hunk 1]
```diff
@@ -15,10 +15,18 @@ internal class CraftTreePatcher
     #region Internal Fields
 
     internal static Dictionary<CraftTree.Type, ModCraftTreeRoot> CustomTrees = new();
-    internal static Dictionary<CraftTree.Type, List<Node>> NodesToRemove = new();
-    internal static Dictionary<CraftTree.Type, List<CraftingNode>> CraftingNodes = new();
-    internal static Dictionary<CraftTree.Type, List<TabNode>> TabNodes = new();
-    internal static Dictionary<CraftTree.Type, CraftTree> CachedTrees = new();
```
Comment (JKohlman) [inline Nautilus/Patchers/CraftTreePatcher.cs:21]: I might be blind but it looks like you remove `CachedTrees` here (not just moved it) and it's used further down on line 93/94.

Comment (JKohlman) [reply Nautilus/Patchers/CraftTreePatcher.cs:21]: Potentially a consequence of just pulling this one commit from amongst your others

Comment (RamuneNeptune) [reply Nautilus/Patchers/CraftTreePatcher.cs:21]: Also somehow turned the Dictionaries into lists.. lol

Comment (MrPurple6411) [reply Nautilus/Patchers/CraftTreePatcher.cs:21]: looks like it reverted parts of my commit from the other day

Comment (JKohlman) [reply Nautilus/Patchers/CraftTreePatcher.cs:21]: Yeah that explains both issues, Ramune didn't have those when they made this commit. CachedTrees didn't exist and they weren't* dictionaries at the point when the commit was made.

Edited to fix 2am brain

Comment (MrPurple6411) [reply Nautilus/Patchers/CraftTreePatcher.cs:21]: I mean they should be dictionaries.

**Review by Metious — APPROVED** [review]

**Pull request merged**